sklearn/multioutput.py

```diff
@@ -0,0 +1,212 @@
+        self._num_y = y.shape[1]
+
+        ## create a dictionary to hold the estimators. 
+        self.estimators_ ={}
+        
+        for i in range(self._num_y):
+            # init a new classifer for each and fit it. 
+            estimator = clone(self.estimator)   #make a fresh clone
+            ovr = OneVsRestClassifier(estimator,self.n_jobs)
+            self.estimators_[i] = ovr.fit(X,y[:, i])
+            
+        return self
+    
+    def predict(self, X):
+        """Predict multi-class multiple target variable using a model
+         trained for each target variable. 
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+        Data.
+
+        Returns
+        -------
+        y : dict of [sparse array-like], shape = {predictors: n_samples}
+          or {predictors: [n_samples, n_classes], n_predictors}.
+            Predicted multi-class targets across multiple predictors.
+            Note:  entirely separate models are generated
+            for each predictor.
+        """
+        # check to see if the fit has been performed
+        check_is_fitted(self, 'estimators_')
+        
+        results = {}
+        for label, model_ in self.estimators_.iteritems():
+            results[label] = model_.predict( X)
+        return(results)
+    
+    def predict_proba(self, X):
+        """Probability estimates. This returns prediction probabilites 
+        for each class for each label in the form of 
+        a dictionary. 
+       
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+        Returns
+        -------
+        prob_dict (dict) A dictionary containing n_label sparse arrays 
+        with shape = [n_samples, n_classes]. 
+        Each row in the array contains the the probability of the 
+        sample for each class in the model,
+        where classes are ordered as they are in `self.classes_`.
+        """
+        # check to see whether the fit has occured. 
+        check_is_fitted(self, 'estimators_')
+        
+        results ={}
+        for label, model_ in self.estimators_.iteritems():
+            results[label] = model_.predict_proba(X)
+        return(results)
+
+    def score(self, X, Y):
+        """"Returns the mean accuracy on the given test data and labels.
+       
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+        Y : (sparse) array-like, shape = [n_samples, p_targets]
+           
+        Returns
+        -------
+        scores (np.array) Array of p_target floats of the mean accuracy 
+        of each estimator_.predict wrt. y.   
+        """
+        check_is_fitted(self, 'estimators_')
+        # Score the results for each function
+        results =[]
+        for i in range(self._num_y):
+            estimator = self.estimators_[i]
+            results.append(estimator.score(X,Y[:,i]))
+        return results
```

Comment:
I am really sorry for not being clear here. It was not the common tests, but the tests written in this PR that were failing, It was due to accessing Y.shape[1] when something like Y[:, i] was used as the input.